backend/grpc-server/src/server/payments.rs

```diff
@@ -252,287 +257,740 @@ impl PaymentOperationsInternal for Payments {
-        let response = external_services::service::execute_connector_processing_step(
-            &self.config.proxy,
-            connector_integration,
-            router_data,
-            None,
-        )
-        .await
-        .switch()
-        .map_err(|e| e.into_grpc_status())?;
+                Ok(tonic::Response::new(authorize_response))
+            }
+            .await;
+        let duration = start_time.elapsed().as_millis();
+        current_span.record("response_time", duration);
 
-        // Generate response
-        let authorize_response = domain_types::types::generate_payment_authorize_response(response)
-            .map_err(|e| e.into_grpc_status())?;
+        match &result {
+            Ok(response) => {
+                current_span.record("response_body", tracing::field::debug(response.get_ref()));
 
-        Ok(tonic::Response::new(authorize_response))
+                let status = response.get_ref().status.to_string();
+                let status_str = attempt_status_to_str(status);
+                current_span.record("flow_specific_fields.status", status_str);
+            }
+            Err(status) => {
+                current_span.record("error_message", status.message());
+                current_span.record("status_code", status.code().to_string());
+            }
+        }
+        result
     }
 
+    #[tracing::instrument(
+        name = "payment_sync",
+        fields(
+            name = crate::consts::NAME,
+            service_name = crate::consts::PAYMENT_SERVICE,
+            service_method = FlowName::Psync.to_string(),
+            request_body = tracing::field::Empty,
+            response_body = tracing::field::Empty,
+            error_message = tracing::field::Empty,
+            merchant_id = tracing::field::Empty,
+            gateway = tracing::field::Empty,
+            request_id = tracing::field::Empty,
+            time_stamp = tracing::field::Empty,
+            status_code = tracing::field::Empty,
+            message = "Golden Log Line (incoming)",
+            response_time = tracing::field::Empty,
+            tenant_id = tracing::field::Empty,
+            flow = FlowName::Psync.to_string(),
+            flow_specific_fields.status = tracing::field::Empty,
+        )
+        skip(self, request)
+    )]
     async fn get(
         &self,
         request: tonic::Request<PaymentServiceGetRequest>,
     ) -> Result<tonic::Response<PaymentServiceGetResponse>, tonic::Status> {
-        self.internal_payment_sync(request).await
+        let current_span = tracing::Span::current();
+        let (gateway, merchant_id, tenant_id, request_id) =
+            connector_merchant_id_tenant_id_request_id_from_metadata(request.metadata())
+                .map_err(|e| e.into_grpc_status())?;
+        let req_body = request.get_ref();
+        let req_body_json =
+            serde_json::to_string(req_body).unwrap_or_else(|_| "<serialization error>".to_string());
```

Comment:
Same here and elsewhere in the code, we're ignoring the error. Is that intentional?